lib/pages/timetable/view/events/event_view.dart

```diff
@@ -73,26 +73,31 @@ class _EventViewState extends State<EventView> {
       actions: [
         AppScaffoldAction(
             icon: Icons.edit,
-            onPressed: () {
-              final user = Provider.of<AuthProvider>(context, listen: false)
-                  .currentUserFromCache;
-              if (user.canAddPublicInfo) {
-                Navigator.of(context).push(MaterialPageRoute<AddEventView>(
-                  builder: (_) => ChangeNotifierProvider<FilterProvider>(
-                    create: (_) => FilterProvider(
-                      defaultDegree: widget.eventInstance.mainEvent.degree,
-                      defaultRelevance:
-                          widget.eventInstance.mainEvent.relevance,
-                    ),
-                    child: AddEventView(
-                      initialEvent: widget.eventInstance.mainEvent,
-                    ),
-                  ),
-                ));
-              } else {
-                AppToast.show(S.of(context).errorPermissionDenied);
-              }
-            })
+            onPressed: !widget.eventInstance.mainEvent.editable
+                ? null // Disable edit button for non-editable events
+                : () {
+                    final user =
+                        Provider.of<AuthProvider>(context, listen: false)
+                            .currentUserFromCache;
+                    if (user.canAddPublicInfo) {
+                      Navigator.of(context)
+                          .push(MaterialPageRoute<AddEventView>(
+                        builder: (_) => ChangeNotifierProvider<FilterProvider>(
+                          create: (_) => FilterProvider(
+                            defaultDegree:
+                                widget.eventInstance.mainEvent.degree,
+                            defaultRelevance:
+                                widget.eventInstance.mainEvent.relevance,
+                          ),
+                          child: AddEventView(
+                            initialEvent: widget.eventInstance.mainEvent,
+                          ),
+                        ),
+                      ));
+                    } else {
+                      AppToast.show(S.of(context).errorPermissionDenied);
```

Comment:
error permission denied toast would be suitable for holidays as well

Reply:
It's not really a permission issue, nobody can edit holidays regardless of their permission level.